drivers/docker/driver.go

```diff
@@ -0,0 +1,1682 @@
+		"mounts": hclspec.NewBlockSet("mounts", hclspec.NewObject(map[string]*hclspec.Spec{
+			"target":   hclspec.NewAttr("target", "string", false),
+			"source":   hclspec.NewAttr("source", "string", false),
+			"readonly": hclspec.NewAttr("readonly", "bool", false),
+			"volume_options": hclspec.NewBlockSet("volume_options", hclspec.NewObject(map[string]*hclspec.Spec{
+				"no_copy": hclspec.NewAttr("no_copy", "bool", false),
+				"labels":  hclspec.NewAttr("labels", "map(string)", false),
+				"driver_config": hclspec.NewBlockSet("driver_config", hclspec.NewObject(map[string]*hclspec.Spec{
+					"name":    hclspec.NewAttr("name", "string", false),
+					"options": hclspec.NewAttr("name", "map(string)", false),
+				})),
+			})),
+		})),
+		"network_aliases": hclspec.NewAttr("network_aliases", "list(string)", false),
+		"network_mode":    hclspec.NewAttr("network_mode", "string", false),
+		"pids_limit":      hclspec.NewAttr("pids_limit", "number", false),
+		"pid_mode":        hclspec.NewAttr("pid_mode", "string", false),
+		"port_map":        hclspec.NewAttr("port_map", "map(number)", false),
+		"privileged":      hclspec.NewAttr("privileged", "bool", false),
+		"readonly_rootfs": hclspec.NewAttr("readonly_rootfs", "bool", false),
+		"security_opt":    hclspec.NewAttr("security_opt", "list(string)", false),
+		"shm_size":        hclspec.NewAttr("shm_size", "number", false),
+		"sysctl":          hclspec.NewAttr("sysctl", "map(string)", false),
+		"tty":             hclspec.NewAttr("tty", "bool", false),
+		"ulimit":          hclspec.NewAttr("ulimit", "map(string)", false),
+		"uts_mode":        hclspec.NewAttr("uts_mode", "string", false),
+		"userns_mode":     hclspec.NewAttr("userns_mode", "string", false),
+		"volumes":         hclspec.NewAttr("volumes", "list(string)", false),
+		"volume_driver":   hclspec.NewAttr("volume_driver", "string", false),
+		"work_dir":        hclspec.NewAttr("work_dir", "string", false),
+	})
+
+	// capabilities is returned by the Capabilities RPC and indicates what
+	// optional features this driver supports
+	capabilities = &drivers.Capabilities{
+		SendSignals: true,
+		Exec:        true,
+		FSIsolation: structs.FSIsolationImage,
+	}
+
+	// createClientsLock is a lock that protects reading/writing global client
+	// variables
+	createClientsLock sync.Mutex
+
+	// client is a docker client with a timeout of 5 minutes. This is for doing
+	// all operations with the docker daemon besides which are not long running
+	// such as creating, killing containers, etc.
+	client *docker.Client
+
+	// waitClient is a docker client with no timeouts. This is used for long
+	// running operations such as waiting on containers and collect stats
+	waitClient *docker.Client
+
+	// healthCheckClient is a docker client with a timeout of 1 minute. This is
+	// necessary to have a shorter timeout than other API or fingerprint calls
+	healthCheckClient *docker.Client
```

Comment:
Since only 1 driver instance is created per plugin the clients and their lock don't have to be global anymore!